File: src/Wasserwacht.DigitalGuardBook.Common/Station.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Wasserwacht.DigitalGuardBook.Common
{
    public class Station
    {
        [Required]
        [MaxLength(250)]
        public string Name { get; set; }
        [MaxLength(500)]
        public string Street { get; set; }
        [MaxLength(20)]
        public string ZipCode { get; set; }
        [MaxLength(100)]
        public string City { get; set; }
        [Phone]
        public string PhoneNumber { get; set; }
    }
}

File: digital-guardbook-ui/src/store/station.ts
import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import type { Station } from '@/models/station'
import apolloClient from '@/plugins/apollo'
import gql from 'graphql-tag'

export const useStationStore = defineStore('station', () => {
  const station = ref<Station>()
  const loading = ref<boolean>(false)

  const logo = computed<URL>(
    () =>
      new URL(`https://meine.wasserwacht.bayern/logogenerator/H.php?kv=${station.value?.name}&ov=`)
  )

  async function fetch() {
    loading.value = true

    const { data } = await apolloClient.query({
      query: gql`
        query {
          station {
            name
            address {
              street
              city
              zipCode
            }
          }
        }
      `
    })

    station.value = (data as any).station

    loading.value = false
  }

  return { station, loading, logo, fetch }
})
